### PR TITLE
docs: Re-structure User Guides for Discoverability

### PR DESCRIPTION
**Pull request by statiraju**
Thanks for submitting a PR to Triton!
Please go the the `Preview` tab above this description box and select the appropriate sub-template:

* [PR description template for Triton Engineers](?expand=1&template=pull_request_template_internal_contrib.md)
* [PR description template for External Contributors](?expand=1&template=pull_request_template_external_contrib.md)

If you already created the PR, please replace this message with one of
* [External contribution template](https://raw.githubusercontent.com/triton-inference-server/server/main/.github/PULL_REQUEST_TEMPLATE/pull_request_template_external_contrib.md)
* [Internal contribution template](https://raw.githubusercontent.com/triton-inference-server/server/main/.github/PULL_REQUEST_TEMPLATE/pull_request_template_internal_contrib.md)

and fill it out.

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -135,51 +141,26 @@
...
-html_last_updated_fmt = ""
-html_additional_files = ["index.html"]
+html_theme = "nvidia_sphinx_theme"
 
 # Add any paths that contain custom static files (such as style sheets) here,
 # relative to this directory. They are copied after the builtin static files,
 # so a file named "default.css" will overwrite the builtin "default.css".
 html_static_path = ["_static"]
-html_css_files = ["custom.css"]
+#html_css_files = ["custom.css"] # Not needed with new theme
```
Comment (oandreeva-nv) [inline docs/conf.py:150]:
```suggestion
```

File: docs/conf.py [hunk 2]
```diff
@@ -135,51 +141,26 @@
...
-      <a href="https://www.nvidia.com/en-us/contact/" target="_blank">Contact</a>""",
-    "repository_url": "https://github.com/triton-inference-server/server",
-    "use_repository_button": True,
+    "collapse_navigation": False,
+    "github_url": "https://github.com/triton-inference-server/server",
+    "switcher": {
+        # use for local testing
+        "json_url": "http://localhost:8888/_static/switcher.json",
+        #"json_url": "https://docs.nvidia.com/deeplearning/triton-inference-server/user-guide/"
+        #"docs/_static/switcher.json",
```
Comment (oandreeva-nv) [inline docs/conf.py:159]:
```suggestion
```

File: docs/_static/switcher.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    },
+    {
+      "name": "2.50 (stable)",
+      "version": "2.50",
+      "url": "https://docs.nvidia.com/deeplearning/triton-inference-server/user-guide/docs/"
+    },
+    {
+      "name": "2.49",
+      "version": "2.49",
+      "url": "https://docs.nvidia.com/deeplearning/triton-inference-server/user-guide/docs/"
```
Comment (rmccorm4) [inline docs/_static/switcher.json:14]: Do we have different urls available for each version?

File: docs/server_guide/state_management.rst [hunk 4]
```diff
@@ -0,0 +1,10 @@
+########
+State Management
```
Comment (rmccorm4) [inline docs/server_guide/state_management.rst:2]: On the rendering, the `State Management` section feels unnecessarily nested. It's a single indirection to the Implicit State Management page with no other info/links.

Maybe the State Management tab/page can also provide a link to the Sequence Batch Scheduler?

File: docs/server_guide/features.rst [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+########
+
+.. toctree::
+    :hidden:
+    :caption: Features
+    :maxdepth: 2
+
+    Model_execution <../user_guide/model_execution.md>
+    Scheduler <../user_guide/scheduler.md>
+    Batcher <../user_guide/batcher.md>
```
Comment (rmccorm4) [inline docs/server_guide/features.rst:12]: I think these could probably be joined into one `Scheduling & Batching` section

```
Scheduling & Batching
- Default Scheduler
- Ensemble Scheduler
- Dynamic Batch Scheduler
- Sequence Batch Scheduler
- Custom Batching
```

File: docs/user_guide/model_execution.md [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+be changed. The following figure shows model execution when model1
+is configured to allow three instances. As shown in the figure, the
+first three model1 inference requests are immediately executed in
+parallel. The fourth model1 inference request must wait until one of
+the first three executions completes before beginning.
+
+![Triton Mult-Model Parallel Execution
+Diagram](images/multi_model_parallel_exec.png)
+
+# Models And Schedulers
```
Comment (rmccorm4) [inline docs/user_guide/model_execution.md:69]: This Model Execution doc feels a bit strange. The concurrent execution section at the top makes sense, but the following sections on schedulers, stateless, control inputs, etc. feel like they don't belong in this category/doc.

**Review by nv-kmcgill53 — APPROVED** [review]

**Pull request merged**